docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/RoundtripDateOnly.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Text.Json;
+
+namespace SystemTextJsonSamples;
+
+public class RoundtripDateOnly
+{
+    public static void Run()
+    {
+        Console.WriteLine("""
+            Serialize SeriesDataPoint class to a JSON string,
+            then deserialize it back object.
+            """);
+
+        // <RoundtripDateOnly>
+        SeriesDataPoint originalData = new(
+            Id: Guid.NewGuid(),
+            Value: 3.791m,
+            Date: new DateOnly(2002, 1, 13),
+            Time: new TimeOnly(5, 3, 1));
+        string serialized = JsonSerializer.Serialize(originalData);
+
+        Console.WriteLine($"Resulting JSON: {serialized}");
+
+        SeriesDataPoint deserializedData =
+            JsonSerializer.Deserialize<SeriesDataPoint>(serialized)!;
+
+        bool valuesAreTheSame = originalData == deserializedData;
+        Console.WriteLine($"""
+            Original record is the same as the deserialized record: {valuesAreTheSame}
+            """);
+        // </RoundtripDateOnly>
+    }
+}
+
+// <SeriesDataPoint>
+sealed file record class SeriesDataPoint(
```

Comment:
I'm not fond of the optional `class` keyword with `record`, as the word `class` implies to me that we'll be getting reference equality. Could make this just a `record` or a `record struct` to emphasize value equality. Just a thought, feel free to ignore.